packages/plugin-hono/test/hono.test.ts

```diff
@@ -0,0 +1,388 @@
+import Client from '@bugsnag/core/client'
+import plugin from '../src/hono'
+import { EventPayload } from '@bugsnag/core'
+import Event from '@bugsnag/core/event'
+
+jest.mock('../src/load-connection-info', () => {
+  return jest.fn().mockResolvedValueOnce({
+    remote: { address: '1.2.3.4', addressType: 'v4', port: '1234' }
+  }).mockResolvedValueOnce(null)
+})
+
+const noop = () => {}
+const id = <T>(a: T) => a
+const logger = () => ({
+  debug: jest.fn(),
+  info: jest.fn(),
+  warn: jest.fn(),
+  error: jest.fn()
+})
+
+describe('plugin: hono', () => {
+  it('exports two middleware functions', () => {
+    const c = new Client({ apiKey: 'api_key', plugins: [plugin] })
+    c._sessionDelegate = {
+      startSession: () => c,
+      pauseSession: () => {},
+      resumeSession: () => c
+    }
+
+    const middleware = c.getPlugin('hono')
+
+    if (!middleware) {
+      throw new Error('getPlugin("hono") failed')
+    }
+
+    expect(typeof middleware.requestHandler).toBe('function')
+    expect(middleware.requestHandler.length).toBe(2)
+    expect(typeof middleware.errorHandler).toBe('function')
+    expect(middleware.errorHandler.length).toBe(2)
+  })
+
+  describe('requestHandler', () => {
+    it('should clone the client, start a session and attach the cloned client to the context', async () => {
+      const client = new Client({ apiKey: 'api_key', plugins: [plugin] })
+
+      const startSession = jest.fn().mockReturnValue(client)
+      const pauseSession = jest.fn()
+      const resumeSession = jest.fn().mockReturnValue(client)
+
+      client._sessionDelegate = { startSession, pauseSession, resumeSession }
+      client._logger = logger()
+      client._clientContext = { run: jest.fn() }
+
+      const middleware = client.getPlugin('hono')
+
+      if (!middleware) {
+        throw new Error('getPlugin("hono") failed')
+      }
+
+      const context = {} as any
+      const next = jest.fn()
+
+      await middleware.requestHandler(context, next)
+
+      expect(client._logger.warn).not.toHaveBeenCalled()
+      expect(client._logger.error).not.toHaveBeenCalled()
+      expect(startSession).toHaveBeenCalledTimes(1)
+      expect(pauseSession).not.toHaveBeenCalled()
+      expect(resumeSession).not.toHaveBeenCalled()
+      expect(context.bugsnag).toStrictEqual(expect.any(Client))
```

Comment:
does this pass with `toStrictEqual(client)`? should this context be a reference to the client we've created?